### ISSUE TITLE
Refreshable Braille device compatibility

### ISSUE THREAD

**Issue by benetechMatt** [benetech/MathShare#947]
Is Mathshare usable with refreshable braille devices? Assuming they are using NVDA on the computer.

1. evaluate usability
2. create issues for necessary changes/additions
3. test with select teachers/students

Aha! Link: https://diagramlabs.aha.io/features/B-374

**Comment by jscholes** [comment]
I don't think this will currently work, for two primary reasons:

1. The necessary infrastructure is not yet available to surface appropriate content for braille users only (e.g. braille-related ARIA attributes).
2. An engine is implemented for speech rules, in terms of translating Math to something users can understand.  But not for braille.

@sinabahram Do you have any more insights about either of the above?

**Comment by sinabahram** [comment]
Ok, a few things. Technically speaking, James is correct about how this should be ultimately done via aria-braillelabel and possibly also aria-brailleroledescription. Currently, what happens is that NVDA does offer speech and Nemeth in Braille via the MathPlayer 4 software being installed on the computer, for MathML fields of course.

The issue is that I don't believe that is the subsystem being used here to generate the speech and Braille. Instead, those speech rules are coming from MathLive or somewhere else? Neil needs to chime in on this. If they are coming from speech rules engine (SRE), which is from MathJax, then I don't know what they are doing about Braille, but I don't think anything because nothing could be done with web technologies until we, the aria working group, finally passed aria-braillelabel and aria-brailleroledescription, which are slated for release in V1.2.

Now, turning our attention over to how Sam solved this problem at Pearson, I believe that he instructs folks to manipulate a few verbosity settings locally (this is unavoidable; for example, key echo), and then he moves focus to a field and back to where focus was, thereby causing the Braille to appear. Neil can chime in more on that one as well.

So, I'm with James in that I'm unclear how we are assuming this thing is going to have Braille output, far before we go down the path of having teachers or students test anything.
